integration/entity.go

```diff
@@ -25,11 +25,14 @@ type Entity struct {
 
 // EntityMetadata stores entity Metadata
 type EntityMetadata struct {
-	Name                 string              `json:"name"`
-	Namespace            string              `json:"type"`          // For compatibility reasons we keep the type.
-	IdentifierAttributes []map[string]string `json:"id_attributes"` // For entity key uniqueness
+	Name      string   `json:"name"`
+	Namespace string   `json:"type"`          // For compatibility reasons we keep the type.
+	IDAttrs   []AttrKV `json:"id_attributes"` // For entity key uniqueness
```

Comment:
I prefer not using a customized type for readability reasons. Using `map[string]string` avoids the knowledge about `AttrKV`. I would go for new type like `AttrKV` for these cases:
- there is a clear method we want to attach to this new type
- we need the type for differentiate (a good example would be when we use values inside `context.Context`